summaries/search-based-survey.md

```diff
@@ -0,0 +1,335 @@
+# Search-Based Software Test Data Generation
+**Article:**
+*Phil McMinn*. Search-Based Software Test Data Generation: A Survey. Software Testing, Verification and Reliability, vol. 14, no. 2, pp. 105–156, 2004.
+
+## Metaheursitic Search Applications
+**Goals:**
+ - Coverage of the code.
+ - Exercise a specific feature of the program.
+ - Disprove grey-box properties.
+ - Verify non-functional properties.
+
+## Metaheursitic Search Techniques
+Solution should be encoded such that they can be manipulated by the search and 'neighbours' can be compared. 
+Using an *objective function*, the search distinguishes *good* solutions from *bad* solutions and guides the search using this function.
+
+### Hill Climbing
+This is a local search algorithm that starts off at a random starting point and tries to look at neighbouring points in the search space to improve the solution.
+It is simple and fast, it however often reaches sub-optimal solutions when there are many local optimal.
+This can be slightly avoided by starting at multiple random starting points.
+
+### Simulated Annealing
+Simulated Annealing is similar to Hill Climbing, but probabilistically accepts poorer solutions.
+Starts off with lots of freedom in movement, but limits its movement as the search progresses.
+
+### Evolutionary Algorithms
+#### Evolution Strategies
+Evolution Strategies use mutation: a process of randomly modifying solutions. 
+
+#### Genetic Algorithms
+In Genetic Algorithms the search is driven by *recombination*, a mechanism of exchange of information between solutions to create new ones.
+Genetic Algorithms keep track of a population of solutions, not only one solution. 
+This allows for the sampling of a larger search space.
+The population is combined a mutated to evolve successive populations.
+Favour 'fitter' solutions to be combined and form even better solutions.
+Choosing 'fitter' solutions too heavily can have a negative effect and reduce diversity and can cause *premature convergence*.
+
+**Fitness-proprotionate selection:**
+The number of times an individual is selected is proportional to its relative fitness.
+Difficulties: maintaining a constant *selective pressure* (grade of selecting different solutions).
+
+**Linear Ranking:**
+Individuals are sorted by fitness and a selective bias of *Z* is applied to each of the individuals.
+This differs from fitness-proportionate selection in the sense that a constant bias is applied and thus the selective pressure is more constant and controlled.
+
+**Tournament Selection:** 
+Two individuals are chosen at random, a random number is chosen and compared to the probability of the better individual being selected.
```

Comment:
Two individuals are chosen at random. And random number is chosen and compared to the probability of the better individual being selected.